Lib/ast.py

```diff
@@ -346,7 +346,7 @@ def __instancecheck__(cls, inst):
             except AttributeError:
                 return False
             else:
-                return type(value) in _const_types[cls]
+                return isinstance(value, _const_types[cls])
```

Comment:
It looks like making this `isinstance` will allow type sub-classes for all the different constant node types.  I guess this only affects 'ast' module users that are using the old class names?  Just wondering if it is safe to allow any subtype in there.  The constants need to get marshaled don't they?

Reply:
I don't think that serialization has to be considered in isinstance().

> It looks like making this isinstance will allow type sub-classes for all the different constant node types.

I don't understand. The second argument is _const_types[cls], not _const_types. ast.Str only check for str and str subclasses.

Reply:
This can be done simpler if `Str.__new__` return `Str` instead of `Constant`. But I think that since old classes will be removed eventually, it is better to return the pure `Constant`.

I don't think we should support `ast.Str({})`.